masonry_core/src/passes/paint.rs

```diff
@@ -39,7 +44,7 @@ fn paint_widget(
         widget_children: widget.children.reborrow_mut(),
         debug_paint,
     };
-    if ctx.widget_state.request_paint {
+    if ctx.widget_state.request_paint && !is_stashed {
```

Comment:
Maybe there should be a debug assert to ensure that we aren't seeing something happen where `request_paint` is true but it was stashed.